### PR TITLE
feat: Support the customized container

### PR DESCRIPTION
**Pull request by LGDoor**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring cloud

**Comment by bowen5** [comment]
Would you prepare a doc like this https://docs.microsoft.com/en-us/cli/azure/spring-cloud?view=azure-cli-latest to guide customer how to use those commands.

**Comment by LGDoor** [comment]
> Would you prepare a doc like this https://docs.microsoft.com/en-us/cli/azure/spring-cloud?view=azure-cli-latest to guide customer how to use those commands.

Already added in the help.

**Review by wangzelin007 — COMMENTED** [review]
May I ask you to add some scenario tests for those new commands?
About `--registry-username <username> --registry-password <password>`, we can use mock test.

File: src/spring-cloud/setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '2.11.2'
+VERSION = '2.11.2-container'
```
[the remark this answers is not in the document]
Comment (LGDoor) [reply src/spring-cloud/setup.py]: Updated.

**Comment by LGDoor** [comment]
> May I ask you to add some scenario tests for those new commands? About `--registry-username <username> --registry-password <password>`, we can use mock test.

Hi, I've added a test case for this. Could you please review again?

**Comment by wangzelin007** [comment]
@LGDoor
Please write the description of changes into HISTORY.rst to release the new extension version.
Others look good to me.

**Review by wangzelin007 — CHANGES_REQUESTED** [review]
@LGDoor Xiangyang Yu FTE
Please write the description of changes into HISTORY.rst to release the new extension version.
Others look good to me.

**Comment by LGDoor** [comment]
> @LGDoor Please write the description of changes into HISTORY.rst to release the new extension version. Others look good to me.

updated

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**